src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -167,22 +169,54 @@ public TypeSymbolWithAnnotations AsNullableReferenceOrValueType(CSharpCompilatio
         /// </summary>
         public abstract TypeSymbolWithAnnotations AsNullableReferenceType();
         public abstract TypeSymbolWithAnnotations AsNotNullableReferenceType();
-        public abstract TypeSymbolWithAnnotations AsObliviousReferenceType();
 
         public abstract TypeSymbolWithAnnotations WithModifiers(ImmutableArray<CustomModifier> customModifiers);
+        protected abstract TypeSymbolWithAnnotations WithNonNullTypes(bool nonNullTypes);
 
         public abstract TypeSymbol TypeSymbol { get; }
         public virtual TypeSymbol NullableUnderlyingTypeOrSelf => TypeSymbol.StrippedType();
 
         /// <summary>
-        /// Is this a nullable reference or value type.
-        /// If it is a nullable value type, <see cref="TypeSymbol"/>
+        /// Returns:
+        /// true if this is a nullable reference or value type;
+        /// false if this is an unannotated reference type and [NonNullTypes(true),
+        /// or a value type regardless of [NonNullTypes]; and
+        /// null if an unannotated reference type and [NonNullTypes(false)].
+        /// If this is a nullable value type, <see cref="TypeSymbol"/>
         /// returns symbol for constructed System.Nullable`1 type.
-        /// If it is a nullable reference type, <see cref="TypeSymbol"/>
+        /// If this is a nullable reference type, <see cref="TypeSymbol"/>
         /// simply returns a symbol for the reference type.
         /// </summary>
         public abstract bool? IsNullable { get; }
 
+        /// <summary>
```

Comment:
This comment seems wrong, as there is no `null` case.
Here's my understanding:
- `string`, `int`, `T` are unannotated
- `string?`, `T? where T:class` are annotated
- `int?` and `T? where T:struct` are also annotated, if I understand properly